x/programs/rust/README.md

```diff
@@ -25,6 +25,100 @@ Just a couple things to note. Serialization is minimal, yet there are certain as
 
 On the Go side, we unmarshal in the same order.
 
+## Storage
+
+A Program has 2 storage types.
+
+- `Memory` in WebAssembly is a linear buffer of unsigned bytes that can read and
+written to by both the guest or host. Because of the very minimal types
```

Comment:
nit: The disambiguation between guest and host is non-obvious in this context. Is the `Program` a guest? Might be worth providing more context.